AU2/plugins/sanity_checks/IncorrectPseudonymFormatter.py

```diff
@@ -0,0 +1,43 @@
+import re
+from typing import List, Dict
+
+from AU2.database.model import Event
+from AU2.html_components import HTMLComponent
+from AU2.plugins.sanity_checks.model.SanityCheck import SanityCheck
+from AU2.plugins.sanity_checks.model.SanityCheck import Suggestion
+
+class IncorrectPseudonymFormatter(SanityCheck):
+
+    identifier = "Incorrect_Pseudonym_Formatter"
+
+    def _fix(self, string: str, fixes: Dict[str, str]):
+        search_pattern = r"\[([0-9]+)\]"
+        for match in re.findall(search_pattern, string):
+            fixes[f"[{match}]"] = f"[P{match}]"

```

Comment:
Imo the way the suggestions are presented isn't very enlightening, since it doesn't show the context where the mistake appears. This could be fixed by including some surrounding text, e.g. using the regex
```regex
(\S*[^\w\n]*)\[([0-9]+)\]([^\w\n]*\S*)
```
(which includes the neighbouring words)

There is an edge case where that doesn't work correctly, namely when we have overlapping matches. But at the very least it raises awareness of the nearby error ¯\\\_(ツ)\_/¯.

```suggestion
    def _fix(self, string: str, fixes: Dict[str, str]):
        search_pattern = r"(\S*[^\w\n]*)\[([0-9]+)\]([^\w\n]*\S*)"
        for match in re.findall(search_pattern, string):
            fixes[f"{match[0]}[{match[1]}]{match[2]}"] = f"{match[0]}[P{match[1]}]{match[2]}"

```

Reply:
Good idea! I'll use a variant of this.